pallets/core/src/util/batch_update.rs

```diff
@@ -0,0 +1,512 @@
+use super::BoundedKeyValue;
```

Comment:
Can you please document this pattern of batch updates with some examples or tests?